File: src/main/java/com/summer/lijiahao/utils/openapi/ui/OpenApiTool.java
package com.summer.lijiahao.utils.openapi.ui;

import com.intellij.openapi.ui.Messages;

import javax.swing.*;
import java.lang.reflect.Field;

public class OpenApiTool {
    private JTextField requestText;
    private JRadioButton newRadio;
    private JRadioButton oldRadio;
    private JPanel sendLab;
    private JPanel mainPanel;
    private JLabel toolVesrion;
    private JLabel requestLab;
    private JPanel receiveLab;
    private JTextArea sendText;
    private JTextArea receiveText;

    private ButtonGroup buttonGroup;

    public OpenApiTool() {
        this.buttonGroup = new ButtonGroup();
        this.buttonGroup.add(this.newRadio);
        this.buttonGroup.add(this.oldRadio);
    }

    public JComponent getComponent() {
        return this.mainPanel;
    }

    public <T> T getComponent(String componentName, Class<T> clazz) {
        try {
            Field declaredField = getClass().getDeclaredField(componentName);
            declaredField.setAccessible(true);
            return (T) declaredField.get(this);
        } catch (Exception e) {
            Messages.showErrorDialog(e.getMessage(), "Error");
            return null;
        }
    }
}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/opanapi/ui/OpenApiTool.kt
package com.summer.lijiahao.nccdevtool.opanapi.ui

import javax.swing.*

class OpenApiTool {
    var requestText: JTextField? = null
    var newRadio: JRadioButton? = null
    var oldRadio: JRadioButton? = null
    var sendLab: JPanel? = null
    var contentPane: JPanel? = null
    var toolVesrion: JLabel? = null
    var requestLab: JLabel? = null
    var receiveLab: JPanel? = null
    var sendText: JTextArea? = null
    var receiveText: JTextArea? = null

    var buttonGroup: ButtonGroup


    init {
        buttonGroup = ButtonGroup()
        buttonGroup.add(newRadio)
        buttonGroup.add(oldRadio)
    }


    fun getComponent(): JComponent {
        return contentPane!!
    }
}